### ISSUE TITLE
Optimize dispatcher references when using Python plugins

### ISSUE THREAD

**Issue by matiasinsaurralde** [TykTechnologies/tyk#2903]
**Branch/Environment/Version**
- Branch/Version: Master
- Environment: All

**Describe the bug**
Before 2.9, we initialized reusable references in `Python_NewDispatcher`, this included not just the dispatcher module but also the `dispatch_hook` method.
This is found [here](https://github.com/TykTechnologies/tyk/blob/release-2.8/coprocess_python.go#L104).

In 2.9, after implementing the dynamic Python loader, a similar behavior was implemented but the `dispatch_hook` reference is retrieved on every `Dispatch` call, on every request:

**Enhancement**
Because method references can be safely reused, we could move this reference to the dispatcher initialization block. This will reduce the number of Python objects in memory and it's also one less `cgo` call.

PR is #2898.

**Comment by christtyk** [comment]
@matiasinsaurralde is it still valid?

**Comment by matiasinsaurralde** [comment]
@christtyk Yes, it's a small improvement that results in less CPython calls, which it's positive 👍

**Comment by christtyk** [comment]
do you know if there's already something in Jira for it before I make an issue? 👍

**Comment by matiasinsaurralde** [comment]
@christtyk no JIRA ticket yet

**Comment by vverbani** [comment]
Created a Jira ticket for this also updated the title to reference the Jira ticket #.
